### PR TITLE
Update release notes for v0.3.1.

### PR DESCRIPTION
**Pull request by briansmith**

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,3 +1,23 @@
...
+
+Conduit 0.3.1 improves Conduit's resilience and transparency.
+
+* Proxy (data plane)
+  * The proxy now makes fewer changes to requests and responses being proxied. In particular, requests and responses without bodies or with empty bodies are better supported.
+  * HTTP/1 requests with different `Host` header fields are no longer sent on the same HTTP/1 connection even when those hostnames resolve to the same IP address.
+  * A connection leak during proxying of non-HTTP TCP connections was fixed.
+  * The proxy now handles unavailable services more gracefully by timing out while waiting for an endpoint to become available for the service.
+* Command-line interface
+  * `$KUBECONFIG` with multiple paths is now supported.
```
Comment (wmorgan) [inline CHANGES.md]: Was this a non-Buoyant contribution? Would be nice to call out GitHub usernames if so.

Comment (briansmith) [reply CHANGES.md]: > Was this a non-Buoyant contribution? Would be nice to call out GitHub usernames if so.

There were two non-Buoyant contributions mentioned in the release notes. I added a reference to the PR and the author to each.

File: CHANGES.md [hunk 2]
```diff
@@ -1,3 +1,23 @@
+## v0.3.1
+
+Conduit 0.3.1 improves Conduit's resilience and transparency.
+
+* Proxy (data plane)
+  * The proxy now makes fewer changes to requests and responses being proxied. In particular, requests and responses without bodies or with empty bodies are better supported.
```
Comment (klingerf) [inline CHANGES.md]: The rest of this file is wrapped at 100 characters -- can I bug you to do that for these long bullet points as well?

Comment (briansmith) [reply CHANGES.md]: > The rest of this file is wrapped at 100 characters -- can I bug you to do that for these long bullet points as well?

Done. (My editor wasn't 100 characters wide when I was editing this originally, so I didn't realize things were wrapped to 100 characters.)

**Pull request merged**